### PR TITLE
User filtering

### LINKED ISSUE #406: User management filtering
- In User Management screen fix the role filtering of users

### PR DESCRIPTION
**Pull request by Prolas**
## Bug 
In the `UserManagementScreen` (Admin) when filtering users by role, there was always a "No such user exists" message.

## Problem
We added the users manually to the database and did never fill the `userRoles` field, but it was used to filter. 
When using `DatabaseSetup`, everything works fine.

## Solution
* Added manually the roles in the database. 
* Added a role `Admin` in `RoleTypes` 

## Other changements
* Moved the filtering logic in the `UserManagementViewModel`
* Small refactoring of `UserManagementScreen`
* Displays the main role of the user (Admin, Pilot, Crew) and not the whole set of `roleTypes`
* Replacement of `BottomBar` with `AdminBottomBar` in `AdminStats`.

close #406

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/screens/admin/UserManagement.kt [hunk 1]
```diff
@@ -58,52 +70,41 @@ fun UserCard(user: User, onUserClick: (String) -> Unit) {
...
-                    fontWeight = FontWeight.Bold,
-                    style = MaterialTheme.typography.bodyLarge)
-                Text(user.roleTypes.joinToString { it.name })
-              }
+    Row(
+        modifier = Modifier.fillMaxWidth().padding(16.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween) {
+          Text(
+              "${user.firstname} ${user.lastname}",
```
Comment (Plouc314) [inline app/src/main/java/ch/epfl/skysync/screens/admin/UserManagement.kt:87]: Nitpick: you can use `user.name()`

File: app/src/main/java/ch/epfl/skysync/viewmodel/UserManagementViewModel.kt [hunk 2]
```diff
@@ -72,6 +76,28 @@ class UserManagementViewModel(
...
+   *
+   * @param query A string to search for in the user's full name. If empty, this criterion is
+   *   ignored.
+   * @param selectedRole An optional role to filter users by. If null, this criterion is ignored.
+   */
+  fun filterByQueryAndRole(query: String, selectedRole: RoleType?) {
+    _filteredUsers.value =
+        _allUsers.value.filter {
+          (query.isEmpty() ||
+              "${it.firstname} ${it.lastname}".contains(query, ignoreCase = true)) &&
```
Comment (Plouc314) [inline app/src/main/java/ch/epfl/skysync/viewmodel/UserManagementViewModel.kt:96]: Same here

**Review by Plouc314 — APPROVED** [review]
LGTM

File: app/src/main/java/ch/epfl/skysync/screens/admin/UserManagement.kt (hunk 1 above)
Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/screens/admin/UserManagement.kt:87]: I will change it

File: app/src/main/java/ch/epfl/skysync/viewmodel/UserManagementViewModel.kt (hunk 2 above)
Comment (Prolas) [reply app/src/main/java/ch/epfl/skysync/viewmodel/UserManagementViewModel.kt:96]: I will change it

**Pull request merged**